packages/components/progress-bar/src/progress-bar.tsx

```diff
@@ -0,0 +1,229 @@
+   */
+  label?:
+    | string
+    | ReactElement
+    | (MessageDescriptor & { values?: Record<string, ReactNode> })
+    | null;
+  /**
+   * Location of the text in relation to the bar.
+   */
+  labelPosition?: 'top' | 'bottom' | 'left' | 'right';
+  /**
+   * The scale of the width for the label, uses values available in the Constraints.Horizontal component.
+   */
+  labelWidth?:
+    | 6
+    | 7
+    | 8
+    | 9
+    | 10
+    | 11
+    | 12
+    | 13
+    | 14
+    | 15
+    | 16
+    | 'scale'
+    | 'auto';
+  /**
+   * Specifies the use of light colors(default) or dark colors.
+   */
+  isInverted?: boolean;
+  /**
+   * Specifies whether the inner bar should have the styles animated.
+   */
+  isAnimated?: boolean;
+  /**
+   * The scale of the height for the bar, also affects the styles of the label.
+   */
+  height?: '10' | '20';
+  /**
+   * The scale of the width for the label, uses values available in the Constraints.Horizontal component.
+   */
+  barWidth?:
+    | 4
+    | 5
+    | 6
+    | 7
+    | 8
+    | 9
+    | 10
+    | 11
+    | 12
+    | 13
+    | 14
+    | 15
+    | 16
+    | 'scale'
+    | 'auto';
+};
+
+const ProgressBarLabel = (
+  props: Pick<
+    TProgressBarProps,
+    'label' | 'labelWidth' | 'isInverted' | 'height' | 'labelPosition'
+  > & { textAlignment: string }
+) => {
+  if (isNil(props.label)) return null;
+
+  const label = props.label.hasOwnProperty('defaultMessage') ? (
+    <FormattedMessage {...(props.label as MessageDescriptor)} />
+  ) : (
+    props.label
+  );
+
+  return (
+    <Constraints.Horizontal max={props.labelWidth}>
+      <div
+        css={css`
+          min-height: ${heightPerScale[props.height ?? '20']};
+          text-align: ${props.textAlignment};
+        `}
+      >
+        {props.height === '10' ? (
+          <Text.Detail tone={props.isInverted ? 'inverted' : undefined}>
+            {label}
+          </Text.Detail>
+        ) : (
+          <Text.Body
+            as="span"
+            tone={props.isInverted ? 'inverted' : undefined}
+            fontWeight="medium"
+          >
+            {label}
+          </Text.Body>
+        )}
+      </div>
+    </Constraints.Horizontal>
+  );
+};
+
+const Bar = (
+  props: Pick<
+    TProgressBarProps,
+    'progress' | 'height' | 'barWidth' | 'isInverted' | 'isAnimated'
+  >
+) => {
+  const BarChildren = () => {
+    return (
+      <div
+        css={getBackgroundBarStyles(props)}
+        role="progressbar"
+        aria-valuenow={props.progress}
+        aria-label="Progress bar"
+        {...filterAriaAttributes(props)}
+      >
+        <div css={getForegroundBarStyles(props)} />
+      </div>
+    );
+  };
+  // We need a special case for 'auto' as a Constraint with `width: auto` will hide the bar
+  if (props.barWidth === 'auto') {
+    return <BarChildren />;
+  }
+  return (
+    <Constraints.Horizontal max={props.barWidth}>
+      <BarChildren />
+    </Constraints.Horizontal>
+  );
+};
```

Comment:
@CarlosCortizasCT I know this is a bit hacky; hoping you might have a better idea for how to handle this.

Reply:
To be honest, I don't think the `auto` value makes sense for this component.

For what I've been reading, `auto` (this is default `width` for any block component) will take as much as space possible from the parent, but it's based on the contents con the element.
The progress bar has no content since we're working only with relative (percentage) widths of the two divs (background and foreground bars), therefore using `auto` does not use any space at all.

My suggestion is to not allow the `auto` value for the `barWidth` property.
We can keep it for the `labelWidth` prop as this will affect the label element which will always have content when rendered, although I don't think it's going to be useful.

References:
* [one](https://developer.mozilla.org/en-US/docs/Web/CSS/width#auto)
* [two](https://www.youtube.com/watch?v=-st14lUQD3U)
* [three](https://stackoverflow.com/questions/17468733/difference-between-width-auto-and-width-100-percent)
* [four](https://stackoverflow.com/questions/28353625/why-does-percentage-width-work-even-if-no-explicit-width-value-given-for-contain/28354270#28354270)
* [five](https://ishadeed.com/article/auto-css/)

Reply:
Lovely - thank you for considering that. I removed it in [717bf33](https://github.com/commercetools/ui-kit/pull/2702/commits/717bf33bf9cdb0bd2faef4f79b0a41f20ec55fee).